plugins/modules/sonic_image_management.py

```diff
@@ -197,7 +229,19 @@ def validate_and_retrieve_params(module, warnings):
             module.fail_json(msg="{0} -> name is required when {0} -> command = {1}".format(params['category'], params['command']))
         if params.get('path'):
             warnings.append("{0} -> path is ignored when {0} -> command = {1}".format(params['category'], params['command']))
-
+    elif params['command'] == 'gpg-key':
+        if not params.get('keyserver') or not params.get('pubkeyid'):
+            module.fail_json(msg="{0} -> keyserver URL and Key ID are required when {0} -> command = {1}".format(params['category'], params['command']))
+    elif params['command'] == 'verify':
+        if not params.get('verifymethod'):
+            module.fail_json(msg="{0} -> verifymethod is required when {0} -> command =verify".format(params['category']))
+        if params.get('verifymethod') == 'gpg':
+            if not params.get('name') or not params.get('signaturefile'):
+                module.fail_json(msg="{0} -> Image name and GPG signature are required when {0} -> command = {1}".format(params['category'], params['command']))
+        else:
+            if not params.get('name') or not params.get('signaturefile') or not params.get('pubkeyfilename'):
+                module.fail_json(
+                    msg="{0} -> Image name, PKI signature and certificate are required when {0} -> command = {1}".format(params['category'], params['command']))
     return params
```

Comment:
For 'verify', since the input is path of the image, please use 'path' option instead of the 'name' option.

Reply:
Addressed.